### PR TITLE
wfnews-1528 show saved location on wfnews map

### PR DESCRIPTION
**Pull request by yzlucas**

### CODE REVIEW

File: client/wfnews-war/src/main/angular/src/app/components/active-wildfire-map/active-wildfire-map.component.ts [hunk 1]
```diff
@@ -467,9 +471,81 @@ export class ActiveWildfireMapComponent implements OnInit, AfterViewInit {
...
+    } else {
+      this.addAllSavedLocationLabels()
+    }
+  }
+
+  private removeAllSavedLocationLabels() {
+    const map = getActiveMap(this.SMK).$viewer.map;
+  
+    // Iterate over the array of markers and remove them from the map
+    this.savedLocationlabelsToShow.forEach(label => {
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/app/components/active-wildfire-map/active-wildfire-map.component.ts:533]: Generally better to use for..of for performance/scope

File: client/wfnews-war/src/main/angular/src/app/components/active-wildfire-map/active-wildfire-map.component.ts [hunk 2]
```diff
@@ -467,9 +471,81 @@ export class ActiveWildfireMapComponent implements OnInit, AfterViewInit {
...
+      map.removeLayer(label);
+    });
+    this.savedLocationlabelsToShow = [];
+  }
+
+  private addAllSavedLocationLabels() {
+    const map = getActiveMap(this.SMK).$viewer.map;
+    if (this.savedLocationlabelsToShow?.length === 0) {
+      this.savedLocationlabels.forEach(label => {
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/app/components/active-wildfire-map/active-wildfire-map.component.ts:542]: As other comment, especially with references to "this"

File: client/wfnews-war/src/main/angular/src/app/components/active-wildfire-map/active-wildfire-map.component.ts [hunk 3]
```diff
@@ -467,9 +471,81 @@ export class ActiveWildfireMapComponent implements OnInit, AfterViewInit {
...
     this.isMapLoaded = true;
+    this.notificationService.getUserNotificationPreferences().then(response =>{
+      const SMK = window['SMK']
+      const map = getActiveMap(this.SMK).$viewer.map;
+
+      map.on('zoomend', () => {
+        this.updateSavedLocationLabelVisibility();
+      });
+      for (const smkMap in SMK.MAP) {
+        if (Object.prototype.hasOwnProperty.call(SMK.MAP, smkMap)){
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/app/components/active-wildfire-map/active-wildfire-map.component.ts:482]: You can replace `Object.prototype.hasOwnProperty.call` with `Object.hasOwn`

File: client/wfnews-war/src/main/angular/src/app/components/active-wildfire-map/active-wildfire-map.component.ts [hunk 4]
```diff
@@ -467,9 +471,81 @@ export class ActiveWildfireMapComponent implements OnInit, AfterViewInit {
     this.selectedLayer = selectedLayer.value as SelectedLayer || 'wildfire-stage-of-control';
     this.onSelectLayer(this.selectedLayer);
     this.isMapLoaded = true;
+    this.notificationService.getUserNotificationPreferences().then(response =>{
+      const SMK = window['SMK']
+      const map = getActiveMap(this.SMK).$viewer.map;
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/app/components/active-wildfire-map/active-wildfire-map.component.ts:476]: Did you mean to use this.SMK or the const above? If you're not using the const remove it

**Review by dhlevi — APPROVED** [review]
A few general cleanup bits to address but otherwise OK

**Comment by yzlucas** [comment]
> A few general cleanup bits to address but otherwise OK

did some clean up as advised. Will merge in

**Pull request merged**